ringbuf/ring.go

```diff
@@ -0,0 +1,114 @@
+package ringbuf
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"runtime"
+	"sync/atomic"
+	"unsafe"
+
+	"github.com/cilium/ebpf/internal/unix"
+)
+
+type ringbufEventRing struct {
+	fd   int
+	prod []byte
+	cons []byte
+	*ringReader
+}
+
+func newRingBufEventRing(mapFD, size int) (*ringbufEventRing, error) {
+	cons, err := unix.Mmap(mapFD, 0, os.Getpagesize(), unix.PROT_READ|unix.PROT_WRITE, unix.MAP_SHARED)
+	if err != nil {
+		unix.Close(mapFD)
```

Comment:
I dropped `ring.fd` completely and fixed this.

I'm still scratching my head whether we'd even need `ringbufMap` in `Reader`... I'll dig this a bit still.

Reply:
> I'm still scratching my head whether we'd even need `ringbufMap` in `Reader`... I'll dig this a bit still.

FYI: I decided to drop it: it seems we do not need to keep a `Clone()` of the map even in case the original map is `Close()`d (epoll still works).